proposals/stf-general-investment.md

```diff
@@ -0,0 +1,263 @@
+
+*Please interpret this question as broadly as you want. A non-conclusive lists of challenge areas we are interested in learning about are:
+managing contributions, addressing security vulnerabilites, project governance, licencing and other legal issues, dependency management, issue backlog, long-term planning/roadmapping.*
+
+- Handling security vulnerabilities at the scale of Nixpkgs
+  - Dealing with vendored or compiled dependencies
+  - Decreasing turnaround time for triaging incidents and deploying patches
+- We have only very basic solutions to automatically verify, enforce, and implement licensing compliance
+- We lose a lot of momentum on friction in the contribution process
+  - There is no systematic onboarding of new contributors with experienced maintainers
+  - We’re lacking a recommendation system for picking reviewers, ownership is often not evident
+  - We have too many communication platforms, most of which cannot be bridged (well or at all)
+    - Many discussions take place where only part of the stakeholders are present
+    - Moderation is not always catching up with traffic, and incidents repeatedly draw a lot of attention
+  - We lack automation for ensuring correctness and avoiding regressions or additional work, and existing automation is slow or incomplete
+- The issue backlog is not practically triageable any more at this point and only keeps growing
+- The ecosystem and community is still hard to navigate for users and contributors alike
+  - There are too many places for documentation, most of which are constantly going out of date and aren't produced from or tested against the source
+- Some critical components are only superficially maintained because their code is hard to approach for most contributors
+- There are many groups involved in building the Nixpkgs/NixOS distribution, but very little coordination between them
+- It's very hard to measure user impact of certain changes
+  - We struggle with deprecating interfaces and thus accumulate maintenance overhead
+  - This limits our ability to do long-term planning, and many things can't be done without breaking changes
+- Overall, we have very little visibility into what things cost in terms of volunteer time and effort
+- We currently don’t know how to self-host large repositories such as Nixpkgs, and we’re stuck to GitHub
+
+## What are possible alternatives to your project and how does your project compare to them? (optional) 300 words
+
+- Guix
+
+  Guix is an early fork of Nix using Guile as a configuration and implementation language, and is the only project comparable to the Nix ecosystem.
+  Guix provides its own software distribution and operating system configuration framework, with free software exclusively.
+  The community is much smaller, the package set is rather focused on scientific applications.
+  Guix pioneered full-source bootstrapping, which is now ported to Nixpkgs.
```

Comment:
```suggestion
  Guix pioneered full-source bootstrapping, which is now being ported to Nixpkgs.
```

Reply:
That'd be news to me. The bootstrap was ported and can successfully build a gcc8 for i686 but it is not actually used for anything yet.